### PR TITLE
bump: Bump spotbugs and plugins versions

### PR DESCRIPTION
**Pull request by franciscodua**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -39,14 +37,9 @@ sourceGenerators.in(Compile) += Def.task {
...
-       |  val dependenciesLocation: Option[String] = ${ivyPaths
-         .in(Compile)
-         .value
-         .ivyHome
-         .fold("Option.empty[String]")(home => s"""Option("$home")""")}
+       |  val dependenciesLocation: String = "${csrCacheDirectory.in(Compile).value}"
        |
-       |}""".stripMargin
-  )
+       |}""".stripMargin)
```
Comment (lolgab) [inline build.sbt]: Align this.

File: build.sbt [hunk 2]
```diff
@@ -26,9 +26,7 @@ enablePlugins(AshScriptPlugin)
 
 sourceGenerators.in(Compile) += Def.task {
   val file = sourceManaged.in(Compile).value / "com" / "codacy" / "tools" / "spotbugs" / "Keys.scala"
-  IO.write(
-    file,
-    s"""|package com.codacy.tools.spotbugs
+  IO.write(file, s"""|package com.codacy.tools.spotbugs
```
Comment (lolgab) [inline build.sbt]: You don't need the  heading `|`:
```suggestion
  IO.write(file, s"""package com.codacy.tools.spotbugs
```

File: build.sbt (hunk 1 above)
Comment (franciscodua) [reply build.sbt]: Not sure what you mean. What's misaligned?

File: src/main/scala/com/codacy/tools/spotbugs/helper/SpotBugsHelper.scala [hunk 3]
```diff
@@ -1,18 +1,29 @@
...
+import edu.umd.cs.findbugs._
 
 import scala.collection.JavaConverters._
 
 case class PluginDefinition(pluginId: String, pluginOrg: String, pluginName: String, pluginVersion: String) {
+  // look for the plugins in the machines local cache (coursier) downloaded during Compile
+  private val cacheFiles =
+    if (File(Keys.dependenciesLocation).exists) {
+      Files
+        .walk(FileSystems.getDefault.getPath(Keys.dependenciesLocation))
```
Comment (lolgab) [inline src/main/scala/com/codacy/tools/spotbugs/helper/SpotBugsHelper.scala]: Can you use `better-files` for this?

File: src/main/resources/docs/patterns.json [hunk 4]
```diff
@@ -760,7 +783,18 @@
   }, {
     "patternId" : "EL_INJECTION",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "CommandInjection"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: In worker this is categorised as InputValidation

File: src/main/resources/docs/patterns.json [hunk 5]
```diff
@@ -1032,27 +1073,32 @@
   }, {
     "patternId" : "HRS_REQUEST_PARAMETER_TO_COOKIE",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "HTTP"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: Previously, InputValidation

File: src/main/resources/docs/patterns.json [hunk 6]
```diff
@@ -2044,11 +2100,13 @@
...
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "Cryptography"
   }, {
     "patternId" : "OBJECT_DESERIALIZATION",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "CommandInjection"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: Previously, Other

File: src/main/resources/docs/patterns.json [hunk 7]
```diff
@@ -2516,7 +2607,8 @@
   }, {
     "patternId" : "SCRIPT_ENGINE_INJECTION",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "CommandInjection"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: Previously, InputValidation

File: src/main/resources/docs/patterns.json [hunk 8]
```diff
@@ -2536,7 +2628,8 @@
   }, {
     "patternId" : "SEAM_LOG_INJECTION",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "CommandInjection"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: Previously, InputValidation

File: src/main/resources/docs/patterns.json [hunk 9]
```diff
@@ -2728,7 +2830,8 @@
   }, {
     "patternId" : "SPEL_INJECTION",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "CommandInjection"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: Previously, InputValidation

File: src/main/resources/docs/patterns.json [hunk 10]
```diff
@@ -2988,19 +3104,23 @@
   }, {
     "patternId" : "STRUTS1_ENDPOINT",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "InputValidation"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: Previously, Other

File: src/main/resources/docs/patterns.json [hunk 11]
```diff
@@ -3232,11 +3364,13 @@
   }, {
     "patternId" : "UNSAFE_HASH_EQUALS",
     "level" : "Error",
-    "category" : "Security"
+    "category" : "Security",
+    "subcategory" : "Other"
```
Comment (franciscodua) [inline src/main/resources/docs/patterns.json]: InsecureModulesLibraries as an alternative?

File: src/main/resources/docs/patterns.json (hunk 4 above)
Comment (lolgab) [reply src/main/resources/docs/patterns.json]: It's not a problem.. Choose the best match 👌

File: src/main/resources/docs/patterns.json (hunk 5 above)
Comment (lolgab) [reply src/main/resources/docs/patterns.json]: I would choose Cookie here

File: build.sbt [hunk 12]
```diff
@@ -26,27 +26,20 @@ enablePlugins(AshScriptPlugin)
...
+                    |  val toolName: String = "spotbugs"
+                    |
+                    |  val spotBugsVersion: String = "$spotBugsVersion"
+                    |  val findsecbugsVersion: String = "$findsecbugsVersion"
+                    |  val sbContribVersion: String = "$sbContribVersion"
+                    |
+                    |  val defaultLinuxInstallLocation: String = "${defaultLinuxInstallLocation.in(Docker).value}"
+                    |  val dependenciesLocation: String = "${csrCacheDirectory.in(Compile).value}"
+                    |
+                    |}""".stripMargin)
```
Comment (lolgab) [inline build.sbt]: Newline here:
```suggestion
                    |}
                    |""".stripMargin)
```

File: src/main/scala/com/codacy/tools/spotbugs/helper/SpotBugsHelper.scala [hunk 13]
```diff
@@ -1,18 +1,29 @@
 package com.codacy.tools.spotbugs.helper
 
+import java.nio.file.{FileSystems, Files}
```
Comment (lolgab) [inline src/main/scala/com/codacy/tools/spotbugs/helper/SpotBugsHelper.scala]: Ideally you should be able to remove these imports

File: src/main/scala/com/codacy/tools/spotbugs/DocumentationGenerator.scala [hunk 14]
```diff
@@ -96,4 +95,85 @@ object DocumentationGenerator {
...
+    ("WEAK", Pattern.Subcategory.InputValidation),
+    ("CIPHER", Pattern.Subcategory.Cryptography),
+    ("ENCRYPT", Pattern.Subcategory.Cryptography),
+    ("RSA", Pattern.Subcategory.Cryptography),
+    ("DESERIALIZATION", Pattern.Subcategory.CommandInjection),
+    ("DES", Pattern.Subcategory.Cryptography),
+    ("FILE", Pattern.Subcategory.FileAccess),
+    ("PREDICTABLE_RANDOM", Pattern.Subcategory.Cryptography),
+    ("ENTITY", Pattern.Subcategory.Visibility)
+  )
```
Comment (lolgab) [inline src/main/scala/com/codacy/tools/spotbugs/DocumentationGenerator.scala]: You can use tuple sugar here:
```suggestion
  private val matchingPatterns = List(
    "XSS" -> Pattern.Subcategory.XSS,
    "ANDROID" -> Pattern.Subcategory.Android,
    "COOKIE" -> Pattern.Subcategory.Cookies,
    "HTTP" -> Pattern.Subcategory.HTTP,
    "SERVLET" -> Pattern.Subcategory.InputValidation,
    "PATH_TRAVERSAL" -> Pattern.Subcategory.FileAccess,
    "SQL_INJECTION" -> Pattern.Subcategory.SQLInjection,
    "INJECTION" -> Pattern.Subcategory.CommandInjection,
    "JAX" -> Pattern.Subcategory.HTTP,
    "XXE" -> Pattern.Subcategory.InputValidation,
    "CSRF" -> Pattern.Subcategory.CSRF,
    "JSP" -> Pattern.Subcategory.InputValidation,
    "UNVALIDATED" -> Pattern.Subcategory.InputValidation,
    "ENDPOINT" -> Pattern.Subcategory.InputValidation,
    "DIGEST" -> Pattern.Subcategory.Cryptography,
    "WEAK" -> Pattern.Subcategory.InputValidation,
    "CIPHER" -> Pattern.Subcategory.Cryptography,
    "ENCRYPT" -> Pattern.Subcategory.Cryptography,
    "RSA" -> Pattern.Subcategory.Cryptography,
    "DESERIALIZATION" -> Pattern.Subcategory.CommandInjection,
    "DES" -> Pattern.Subcategory.Cryptography,
    "FILE" -> Pattern.Subcategory.FileAccess,
    "PREDICTABLE_RANDOM" -> Pattern.Subcategory.Cryptography,
    "ENTITY" -> Pattern.Subcategory.Visibility
  )
```

File: src/main/scala/com/codacy/tools/spotbugs/helper/SpotBugsHelper.scala (hunk 3 above)
Comment (franciscodua) [reply src/main/scala/com/codacy/tools/spotbugs/helper/SpotBugsHelper.scala]: Done ✅ . Way better

**Pull request merged**